include/binary_graph_stream.h

```diff
@@ -0,0 +1,58 @@
+#include <fstream>
+#include <cstring>
+#include "graph.h"
+
+// A class for reading from a binary graph stream
+class BinaryGraphStream {
+public:
+    BinaryGraphStream(std::string file_name, uint32_t _b) {
+        bin_file.open(file_name.c_str(), std::ios_base::in | std::ios_base::binary);
+
+        // set the buffer size to be a multiple of an edge size and malloc memory
+        buf_size = _b - (_b % edge_size);
+	buf = (char *) malloc(buf_size * sizeof(char));
+	start_buf = buf;
+
+	// read header from the input file
+	bin_file.read(reinterpret_cast<char *>(&num_nodes), 4);
+	bin_file.read(reinterpret_cast<char *>(&num_edges), 8);
+        
+	read_data(); // read in the first block of data
+    }
+    ~BinaryGraphStream() {
+        free(start_buf);
+    }
+    inline uint32_t nodes() {return num_nodes;}
+    inline uint64_t edges() {return num_edges;}
+
+    inline GraphUpdate get_edge() {
+        UpdateType u = (UpdateType) *buf;
+	buf++;
+
+	uint32_t a;
+	std::memcpy(&a, buf, sizeof(uint32_t));
+	buf += 4;
+	uint32_t b;
+	std::memcpy(&b, buf, sizeof(uint32_t));
```

Comment:
u can probably do `memcpy(buf+4)` and increment by 8

Reply:
Will do